docs/3.0rc/deploy/infrastructure-concepts/deploy-via-python.mdx

```diff
@@ -0,0 +1,385 @@
+---
+title: Deploy flows with Python
+description: Learn how to use the Python SDK to deploy flows to run in work pools.
+---
+
+Prefect offers a flexible way to deploy flows to dynamic infrastructure using the Python SDK. 
+This approach allows you to target specific work pools and utilize dynamically provisioned infrastructure.
+
+Deploying flows to dynamic infrastructure offers a flexible and programmatic approach to managing your 
+Prefect workflows. While `flow.serve` remains a valid and straightforward method for deploying flows to persistent 
+infrastructure, there are scenarios where you might need to leverage dynamic, scalable resources. In these cases, 
+`flow.deploy` provides a powerful alternative that allows you to target specific work pools and utilize dynamically 
+provisioned infrastructure.
+
+## When to consider `flow.deploy` over `flow.serve`
+
+The `flow.serve` method is an excellent choice for many use cases, especially when you have readily available, persistent infrastructure. However, you might want to consider using `flow.deploy` in the following situations:
+
+1. Cost optimization: Dynamic infrastructure can help reduce costs by scaling resources up or down based on demand.
```

Comment:
I feel like these could be combined into one bullet point on dynamic infrastructure and you could add other bullet points for other situations. For example:

1. Guardrails for other workflow authors in your organization through customized job templates.
2. Scale and redundancy for large, potentially long-running, workloads with K8s

Reply:
I chose these points because I thought they mapped well onto what users would care about when deciding. I'd prefer not to collapse these ideas since I think they're important to communicate.